### ISSUE TITLE
[Bug] Inconsistent handling of empty channel tabs

### ISSUE THREAD

**Issue by absidue** [iv-org/invidious#3814]
**Describe the bug**
A channel tab being empty is handled differently depending on the tab. The playlists tab displays nothing on the website and returns an empty API response `{"playlists":[],"continuation":null}`, the community tab however shows `This channel hasn't posted yet` on the website and returns a HTTP 500 API response with this error: `{"error":"This channel hasn't posted yet"}`.

It's weird that Invidious treats the playlists tab being empty as okay, but the community tab being empty as an error. The `This channel hasn't posted yet` is displayed on YouTube too, however Invidious happily ignores the `This channel has no playlists.` message on the playlists tab.

**Steps to Reproduce**
Webpage:
1. Visit `{instance}/channel/UCUKPG1-r4WFyxpyhIuCs86Q/playlists`
2. Visit `{instance}/channel/UCUKPG1-r4WFyxpyhIuCs86Q/community`
3. Compare the pages

API:
1. Visit `{instance}/api/v1/channels/UCUKPG1-r4WFyxpyhIuCs86Q/playlists`
2. Visit `{instance}/api/v1/channels/UCUKPG1-r4WFyxpyhIuCs86Q/community`
3. Compare the responses

**Screenshots**
Playlists tab:
![web-playlists](https://github.com/iv-org/invidious/assets/48293849/798666cc-5ea4-4113-96e7-30081bfc8ff2)

Community tab:
![web-community](https://github.com/iv-org/invidious/assets/48293849/80acab24-54e8-417c-8ab6-1591b306aae2)

**Comment by SamantazFox** [comment]
For the community tab, that's an "error" returned by youtube's API. In addition, the community tab still relies on older code, which hasn't been cleaned, so it isn't handled the same way as the others.

**Comment by absidue** [comment]
If that error is expected, does that mean you'll be returning errors for the other channel tab endpoints too or is the goal to have the community tab API endpoint behave the same as the other ones and return an empty response?

Ideally from an API user's perspective, it would be best to only return errors if there is an actual error, an empty channel tab is normal on YouTube, so shouldn't return an error. Would also align it with the behaviour of other endpoints e.g. you don't return an error when there are no search results for a search query.

**Comment by SamantazFox** [comment]
Yes, I agree with you here: the community endpoint shouldn't give an error if there is no posts. Just an empty array.

**Comment by syeopite** [comment]
I'm not sure if I agree with just returning an empty array. Invidious should display some kind of message when the tab is empty instead of just having it be blank imo.

**Comment by SamantazFox** [comment]
@syeopite We were talking about the API. Regarding the frontend, I agree with you!

**Comment by ChunkyProgrammer** [comment]
Playlist tab doesn't seem to be showing anymore for that channel so I guess YouTube fixed that?
